syntax = "proto3";
package auth;

service AuthService {
  rpc ValidateUser (ValidateUserRequest) returns (ValidateUserResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc Register (RegisterRequest) returns (LoginResponse);
};

message ValidateUserRequest{
  string email = 1;
  string password = 2;
};

message ValidateUserResponse{
  bool is_valid = 1;
  User user = 2;
};

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  string access_token = 1;
  User user = 2;
}

message RegisterRequest {
  string email = 1;
  string password = 2;
  string full_name = 3;
  string avatar_url = 4;
}

message User {
  string id = 1;
  string email = 2;
  string full_name = 3;
  string avatar_url = 4;
}